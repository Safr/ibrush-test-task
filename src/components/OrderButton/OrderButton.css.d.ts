export const OrderBtn: string;
export const orderBtn: string;
