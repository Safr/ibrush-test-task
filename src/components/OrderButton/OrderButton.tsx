import React from 'react';
import classes from './OrderButton.css';

interface Props {
  onFetchOrders: () => void;
}

const OrderButton: React.SFC<Props> = ({ onFetchOrders }) => {
  return (
    <button
      className={classes.OrderBtn}
      onClick={onFetchOrders}
    >
     Взять ордеры
    </button>
  );
};

export default OrderButton;
