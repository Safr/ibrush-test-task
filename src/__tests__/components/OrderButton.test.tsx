import React from 'react';
import { shallow } from 'enzyme';
import OrderButton from '../../components/OrderButton/OrderButton';

describe('OrderButton Component', () => {
  let wrapper;
  let onFetchOrders;
  beforeEach(() => {
    onFetchOrders = jest.fn();
    wrapper = shallow(
      <OrderButton
        onFetchOrders={onFetchOrders}
      />,
    );
  });
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should handle onFetchOrders', () => {
    wrapper.find('button').simulate('click');
    expect(onFetchOrders).toHaveBeenCalled();
  });
});
