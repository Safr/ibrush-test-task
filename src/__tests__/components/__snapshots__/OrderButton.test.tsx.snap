// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderButton Component should render correctly 1`] = `
<button
  onClick={[MockFunction]}
>
  Взять ордеры
</button>
`;
